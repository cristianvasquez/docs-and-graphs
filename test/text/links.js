import { expect } from 'expect'

import toMatchSnapshot from 'expect-mocha-snapshot'
import {
  parseWikilink, parseExternalLinks
} from '../../src/text/links.js'

const markdown = [
  'no link',
  '[[link 1]]',
  '[[/full/path/link 1]]',
  '[[/full/path/link 1.md]]',
  '[Alias 2](http://example.com)',
  'some text http://example.com more text',
  '[Alias 3](https://example.com)',
  'some text2 https://example.com more text',
  'https://example.com',
  'http://example.com',
  '[[link 1 |Alias 1]]',
  '![[link 1]]',
  '![[image.png]]',
  '[protocol unknown](hello)',
  '[protocol unknown](file://)',
  '<http://example.org>',
  '<https://example.org>'
]

expect.extend({ toMatchSnapshot })

describe('parseWikilink', async function () {
  for (const current of markdown) {
    it(current, async function () {
      const result = parseWikilink(current)
      expect(result).toMatchSnapshot(this)
    })
  }
})

describe('parseExternalLinks', async function () {
  for (const current of markdown) {
    it(current, async function () {
      const result = parseExternalLinks(current)
      expect(result).toMatchSnapshot(this)
    })
  }
})
