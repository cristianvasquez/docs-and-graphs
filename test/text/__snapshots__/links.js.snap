// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseExternalLinks ![[image.png]] 1`] = `undefined`;

exports[`parseExternalLinks ![[link 1]] 1`] = `undefined`;

exports[`parseExternalLinks [[/full/path/link 1.md]] 1`] = `undefined`;

exports[`parseExternalLinks [[/full/path/link 1]] 1`] = `undefined`;

exports[`parseExternalLinks [[link 1 |Alias 1]] 1`] = `undefined`;

exports[`parseExternalLinks [[link 1]] 1`] = `undefined`;

exports[`parseExternalLinks [Alias 2](http://example.com) 1`] = `
Object {
  "alias": "Alias 2",
  "value": "http://example.com",
}
`;

exports[`parseExternalLinks [Alias 3](https://example.com) 1`] = `
Object {
  "alias": "Alias 3",
  "value": "https://example.com",
}
`;

exports[`parseExternalLinks [protocol unknown](file://) 1`] = `undefined`;

exports[`parseExternalLinks [protocol unknown](hello) 1`] = `undefined`;

exports[`parseExternalLinks http://example.com 1`] = `
Object {
  "alias": undefined,
  "value": "http://example.com",
}
`;

exports[`parseExternalLinks https://example.com 1`] = `
Object {
  "alias": undefined,
  "value": "https://example.com",
}
`;

exports[`parseExternalLinks no link 1`] = `undefined`;

exports[`parseExternalLinks some text http://example.com more text 1`] = `undefined`;

exports[`parseExternalLinks some text2 https://example.com more text 1`] = `undefined`;

exports[`parseWikilink ![[image.png]] 1`] = `
Object {
  "alias": undefined,
  "transclude": true,
  "value": "image.png",
}
`;

exports[`parseWikilink ![[link 1]] 1`] = `
Object {
  "alias": undefined,
  "transclude": true,
  "value": "link 1",
}
`;

exports[`parseWikilink [[/full/path/link 1.md]] 1`] = `
Object {
  "alias": undefined,
  "value": "/full/path/link 1.md",
}
`;

exports[`parseWikilink [[/full/path/link 1]] 1`] = `
Object {
  "alias": undefined,
  "value": "/full/path/link 1",
}
`;

exports[`parseWikilink [[link 1 |Alias 1]] 1`] = `
Object {
  "alias": "Alias 1",
  "value": "link 1 ",
}
`;

exports[`parseWikilink [[link 1]] 1`] = `
Object {
  "alias": undefined,
  "value": "link 1",
}
`;

exports[`parseWikilink [Alias 2](http://example.com) 1`] = `undefined`;

exports[`parseWikilink [Alias 3](https://example.com) 1`] = `undefined`;

exports[`parseWikilink [protocol unknown](file://) 1`] = `undefined`;

exports[`parseWikilink [protocol unknown](hello) 1`] = `undefined`;

exports[`parseWikilink http://example.com 1`] = `undefined`;

exports[`parseWikilink https://example.com 1`] = `undefined`;

exports[`parseWikilink no link 1`] = `undefined`;

exports[`parseWikilink some text http://example.com more text 1`] = `undefined`;

exports[`parseWikilink some text2 https://example.com more text 1`] = `undefined`;
